### DISCUSSION TITLE
recording the result of a transmitting custum message

### DISCUSSION THREAD

**Discussion by michael89-sys** [riebl/artery#193] [General]
hi @riebl , thank you for helping me in the following issue 

[https://github.com/riebl/artery/issues/186](url)

i want to record the element of my custum message (reception and transmission) , i have no idea who to do it , i tried cOutVector by adding just for testing :

 **_Ve.setName("vehicle number : ");
 Ve.record(id);_**

i don't know if what i did is correct even if it's correct , i don't know how to display the results , (i don't know the name of the signals to use them as a source in the ned file
![1](https://user-images.githubusercontent.com/83643634/117670945-a3766700-b1a8-11eb-9977-43f8e62ef7d0.PNG)
![2](https://user-images.githubusercontent.com/83643634/117670949-a40efd80-b1a8-11eb-81c0-4f8a7d97c822.PNG)
![3](https://user-images.githubusercontent.com/83643634/117670953-a40efd80-b1a8-11eb-9f65-2b3c9ab830af.PNG)

)
 any idea how to show them 
Your help will be highly appreciated.

**Comment by riebl** [comment]
`cOutVector` only writes to a data log file, i.e. you don't see anything in the OMNeT++ QtEnv GUI. You can of course visualise the recorded vector by plotting your data after the simulation run.
Instead of dealing manually with `cOutVector`, you can also use OMNeT++ signals along with `@statistic` in NED files. I suggest getting familiar with OMNeT++'s `@statistic` and `@signal` first. The OMNeT++ manual is of great help: https://doc.omnetpp.org/omnetpp/manual/#sec:simple-modules:signals and https://doc.omnetpp.org/omnetpp/manual/#sec:simple-modules:signal-based-statistics

**Comment by michael89-sys** [comment]
i've read and tried different things and i'm steal stuck in this part , 


i added to my trigger() function this line :

    emit(scSignalVeSent, long(packet->getactualBatteryCapacity()));

and in the ned file of my service :

        @signal[VeSent](type=long);
        @statistic[transmission](source=VeSent; record=vector?);

my simulation runs but no result displayed .

**Reply by riebl** [reply]
I am afraid I don't understand your issue quite right: Where do you expect something to be displayed?

**Reply by michael89-sys** [reply]
in my .anf file , (for CAM message i can display for exemple speed heading as a vectors , now i want to display just the vector of "actualBatteryCapacity")

![xx](https://user-images.githubusercontent.com/83643634/118397805-2a6a8a00-b656-11eb-9bcd-7b7bc08a5c70.PNG)

**Reply by riebl** [reply]
Okay, now I understand your issue. Please make sure that your service is actually running, i.e. your *services.xml* deploys your service on the vehicles. In the module tree-view of OMNeT++'s QtEnv, you should see your service as a sub-module of vehicles' *middleware* module.

Also, note that `record=vector?` does not enable recording of this vector but the `?` makes it only a candidate for recording. For testing purposes now I suggest to omit the `?` to keep the number of possible misconfiguration options low.
